examples/mpk.rs

```diff
@@ -0,0 +1,229 @@
+//! This example demonstrates:
+//! - how to enable memory protection keys (MPK) in a Wasmtime embedding (see
+//!   [`build_engine`])
+//! - the expected memory compression from using MPK: it will probe the system
+//!   by creating larger and larger memory pools until system memory is
+//!   exhausted (see [`probe_engine_size`]). Then, it prints a comparison of the
+//!   memory used in both the MPK enabled and MPK disabled configurations.
+//!
+//! You can execute this example with:
+//!
+//! ```console
+//! $ cargo run --example mpk
+//! ```
+//!
+//! Append `-- --help` for details about the configuring the memory size of the
+//! pool. Also, to inspect interesting configuration values used for
+//! constructing the pool, turn on logging:
+//!
+//! ```console
+//! $ RUST_LOG=debug cargo run --example mpk -- --memory-size 512MiB
+//! ```
+//!
+//! Note that OS limits on the number of virtual memory areas (VMAs) can
+//! significantly restrict the total number MPK-striped memory slots; each
+//! MPK-protected slot ends up using a new VMA entry. On Linux, one can raise
+//! this limit:
+//!
+//! ```console
+//! $ sysctl vm.max_map_count
+//! 65530
+//! $ sysctl vm.max_map_count=$LARGER_LIMIT
+//! ```
+
+use anyhow::{anyhow, Result};
+use bytesize::ByteSize;
+use clap::Parser;
+use log::{info, warn};
+use std::str::FromStr;
+use wasmtime::*;
+
+fn main() -> Result<()> {
+    drop(env_logger::try_init());
```

Comment:
Since this is the start of the program this can actually be `env_logger::init()`